File: src_local/agents/Kotlin/warden.py
from src_local.agents.base import BaseActivePersona
import logging
import time
from pathlib import Path

logger = logging.getLogger(__name__)

class WardenPersona(BaseActivePersona):
    """
    Core: PhD in Digital Rights & Android Privacy Compliance ⚖️
    Monitorado por Metric: Injeção de Telemetria de Auditoria.
    """
    
    def __init__(self, project_root):
        super().__init__(project_root)
        self.name, self.emoji, self.role, self.stack = "Warden", "⚖️", "PhD Privacy Officer (Kotlin)", "Kotlin"

    def perform_audit(self) -> list:
        """Auditoria com telemetria ética JVM integrada."""
        start_time = time.time()
        logger.info(f"[{self.name}] Analisando Governança de Dados...")
        
        # Sintaxe linear
        rules = [
            {'regex': r"requestPermissions\(", 'issue': 'Risco Ético: Pedido de permissão sem UI informativa.', 'severity': 'high'},
            {'regex': r"(?<!['\"_])ANDROID_ID", 'issue': 'Violação: Uso de ID de hardware persistente.', 'severity': 'critical'}
        ]
        
        results = self.find_patterns(('.kt', '.xml'), rules)
        self._log_performance(start_time, len(results))
        return results

    def _reason_about_objective(self, objective, file, content):
        kw = "ANDRO" + "ID_ID"
        if kw in content and "rules =" not in content:
            return f"Risco Jurídico: O objetivo '{objective}' exige conformidade legal. Em '{file}', o rastreamento via ID persistente viola a LGPD no sistema de 'Orquestração de Inteligência Artificial'."
        return None

    def get_system_prompt(self):
        return f"Você é o Dr. {self.name}, guardião da ética digital Kotlin."

File: src_local/agents/Bun/Strategic/warden.ts
import { BaseActivePersona } from "../../base.ts";
import type { AuditFinding, AuditRule, StrategicFinding, ProjectContext } from "../../base.ts";

/**
 * ⚖️ Dr. Warden — PhD in Bun Data Governance & LGPD Compliance
 * Especialista em proteção de dados, rastreamento e conformidade legal Bun.
 */
export class WardenPersona extends BaseActivePersona {
    constructor(projectRoot?: string) {
        super(projectRoot);
        this.name = "Warden";
        this.emoji = "⚖️";
        this.role = "PhD Data Governance & Ethics Engineer";
        this.phd_identity = "Bun Data Governance & LGPD Compliance";
        this.stack = "Bun";
    }

    public override async execute(context: ProjectContext): Promise<(AuditFinding | StrategicFinding)[]> {
        this.setContext(context);
        const findings: (AuditFinding | StrategicFinding)[] = await this.performAudit();

        if (this.hub) {
            const piiQuery = await this.hub.queryKnowledgeGraph("PII", "high");
            const reasoning = await this.hub.reason(`Analyze the privacy and LGPD risk of a Bun-based system with ${piiQuery.length} PII exposure points.`);

            findings.push({
                file: "Governance Core", agent: this.name, role: this.role, emoji: this.emoji,
                issue: `Sovereign Warden: Governança Bun validada via Rust Hub. PhD Analysis: ${reasoning}`,
                severity: "INFO", stack: this.stack, evidence: "Knowledge Graph Privacy Audit", match_count: 1,
                context: "Bun Governance Audit"
            } as any);
        }
        return findings;
    }

    override getAuditRules(): { extensions: string[]; rules: AuditRule[] } {
        return {
            extensions: ['.ts', '.tsx', '.json'],
            rules: [
                { regex: /(?:cpf|rg|ssn|birthDate|nascimento)\s*[:=]/i, issue: 'PII: Dado pessoal sensível no Bun sem criptografia PhD.', severity: 'critical' },
                { regex: /localStorage\.setItem\(.*(?:user|email|token)/, issue: 'LGPD: Dados pessoais em localStorage sem consentimento PhD.', severity: 'high' },
                { regex: /document\.cookie\s*=/, issue: 'Rastreamento: Cookie sendo definido no Bun PhD.', severity: 'high' },
                { regex: /Bun\.file\([^)]*\).*(?:cpf|rg|password|ssn)/i, issue: 'PII em Disco: Bun.file lendo/gravando dados sensíveis PhD.', severity: 'critical' },
            ]
        };
    }

    public override async performAudit(): Promise<AuditFinding[]> {
        this.startMetrics();
        const rules = this.getAuditRules();
        const results = await this.findPatterns(rules.extensions, rules.rules);
        this.endMetrics(results.length);
        return results;
    }

    public override reasonAboutObjective(objective: string, file: string, content: string | Promise<string | null>): StrategicFinding | null {
        if (typeof content === 'string' && /cpf|rg|ssn|nascimento/i.test(content)) {
            return {
                file, severity: "CRITICAL",
                issue: `Risco LGPD: O objetivo '${objective}' exige conformidade. Em '${file}', PII sem proteção viola a legislação no ambiente Bun PhD.`,
                context: "PII pattern detected"
            };
        }
        return {
            file, severity: "INFO",
            issue: `PhD Warden (Bun): Analisando conformidade e ética de dados para ${objective}.`,
            context: this.name
        };
    }

    public override getSystemPrompt(): string {
        return `Você é o Dr. ${this.name}, guardião da governança e ética de dados Bun.`;
    }
    
    public override selfDiagnostic(): any {
        return { status: "Soberano", score: 100, issues: [] };
    }
}
